test/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedControllerSpec.scala

```diff
@@ -40,28 +40,31 @@ class ReturnSubmittedControllerSpec extends SpecBase {
   )
 
   "onPageLoad" should {
-    "return OK and the correct view" in forAll { (chargeReference: String, obligationDetails: ObligationDetails) =>
-      implicit val authRequest: AuthorisedRequest[AnyContentAsEmpty.type] =
-        AuthorisedRequest(fakeRequest, internalId, eclRegistrationReference)
-      implicit val messages: Messages                                     = messagesApi.preferred(authRequest)
+    "return OK and the correct view" in forAll {
+      (chargeReference: String, obligationDetails: ObligationDetails, amountDue: BigDecimal) =>
+        implicit val authRequest: AuthorisedRequest[AnyContentAsEmpty.type] =
+          AuthorisedRequest(fakeRequest, internalId, eclRegistrationReference)
+        implicit val messages: Messages                                     = messagesApi.preferred(authRequest)
 
-      val result: Future[Result] =
-        controller.onPageLoad()(
-          fakeRequest.withSession(
-            (SessionKeys.ChargeReference, chargeReference),
-            (SessionKeys.ObligationDetails, Json.toJson(obligationDetails).toString())
+        val result: Future[Result] =
+          controller.onPageLoad()(
+            fakeRequest.withSession(
+              (SessionKeys.ChargeReference, chargeReference),
+              (SessionKeys.ObligationDetails, Json.toJson(obligationDetails).toString()),
+              (SessionKeys.AmountDue, amountDue.toString())
+            )
           )
-        )
 
-      status(result) shouldBe OK
+        status(result) shouldBe OK
 
-      contentAsString(result) shouldBe view(
-        chargeReference,
-        ViewUtils.formatToday(),
-        ViewUtils.formatLocalDate(obligationDetails.inboundCorrespondenceDueDate),
-        obligationDetails.inboundCorrespondenceFromDate.getYear.toString,
-        obligationDetails.inboundCorrespondenceToDate.getYear.toString
-      )(authRequest, messages).toString
+        contentAsString(result) shouldBe view(
+          chargeReference,
+          ViewUtils.formatToday(),
+          ViewUtils.formatLocalDate(obligationDetails.inboundCorrespondenceDueDate),
+          obligationDetails.inboundCorrespondenceFromDate.getYear.toString,
+          obligationDetails.inboundCorrespondenceToDate.getYear.toString,
+          amountDue
+        )(authRequest, messages).toString()
     }
 
```

Comment:
Test missing for IllegalStateException when the amount due is not found in session